pandas/core/strings/object_array.py

```diff
@@ -434,7 +434,7 @@ def _str_get_dummies(self, sep: str = "|", dtype: NpDtype | None = None):
             dummies_dtype = _dtype
         else:
             dummies_dtype = np.bool_
-        dummies = np.empty((len(arr), len(tags2)), dtype=dummies_dtype)
+        dummies = np.empty((len(arr), len(tags2)), dtype=dummies_dtype, order="F")
```

Comment:
@aaronchucarroll @rhshadrach do you remember why this change was included?

Reply:
Ah I see that is based on a comment from @mroeschke at https://github.com/pandas-dev/pandas/pull/59577#discussion_r1750984888